server/tests/security/test_allowlist_bypass.py

```
@@ -0,0 +1,132 @@
+    def test_extra_whitespace_still_blocked(self, cmd):
+        assert any_layer_blocks(cmd), f"Whitespace evasion not caught: {cmd!r}"
+
+
+# ---------------------------------------------------------------------------
+# 3. Deny-list coverage for universal rules
+# ---------------------------------------------------------------------------
+
+class TestUniversalDenyRules:
+    """Every pattern in _UNIVERSAL_DENY_RULES should fire on its canonical example."""
+
+    @pytest.mark.parametrize("cmd,desc", [
```

Ruff (0.15.12)

[warning] Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)